cmake/ShogunFindLAPACK.cmake

```diff
@@ -66,14 +66,14 @@ IF (LAPACK_FOUND)
     ENDIF()
 
     IF (ENABLE_EIGEN_LAPACK)
-      FIND_PATH(MKL_INCLUDE_DIR mkl.h)
+      FIND_PATH(MKL_INCLUDE_DIR mkl.h HINTS ${MKL_PATH}/../include/)
       IF(NOT MKL_INCLUDE_DIR)
         MESSAGE(STATUS "Found MKL, but not mkl.h. Make sure that mkl headers are available in order to use MKL as BLAS/Lapack backend for Eigen.")
         SET(ENABLE_EIGEN_LAPACK OFF CACHE BOOL "Use ${ENABLE_EIGEN_LAPACK}" FORCE)
       ELSE()
         MESSAGE(STATUS "Enabling MKL as BLAS/Lapack backend for Eigen.")
         SET(EIGEN_USE_MKL_ALL 1)
-        target_include_directories(shogun PUBLIC ${MKL_INCLUDE_DIR})
+        target_include_directories(libshogun PUBLIC ${MKL_INCLUDE_DIR})
```

Comment:
@vigsterkr I think this is bug right?

Reply:
we should use the public for shogun when mkl is used because then when you link with shogun.so you need to force that the user links against mkl as well

Reply:
So need to have
```
target_include_directories(shogun PUBLIC ${MKL_INCLUDE_DIR})
target_include_directories(libshogun PUBLIC ${MKL_INCLUDE_DIR})
```
?

Reply:
i think the includes could be `PRIVATE` as i dont think we expose mkl things in our headers.

Reply:
or do we? i'm not so sure.... need to see if any of the headers of shogun exposes mkl in some indirect way..... but definitely libshogun needs that as well

Reply:
well can do a bit of trial and error... at least its working now. Maybe can merge #4523 to see if it works with this CI config?

Reply:
so seems like libshogun can be private, but not shogun. Guess that was expected..